montepy/mcnp_problem.py

```diff
@@ -390,25 +391,56 @@ def add_cell_children_to_problem(self):
         self._transforms = Transforms(transforms)
         self._data_inputs = sorted(set(self._data_inputs + materials + transforms))
 
-    def write_to_file(self, new_problem, overwrite=False):
+    def write_problem(self, destination, overwrite=False):
+        """
+        Write the problem to a file or writeable object.
+
```

Comment:
I'm not sure if this is the most pythonic name.

Reply:
* Openpyxl: `save()` not a fan. This is very MS office centric
* pandas: `to_*()` e.g., `to_excel` `to_json`... I think this makes sense for something so filetype agnostic
* numpy: `tofile()`, `dump`
* openmc: `export_to_xml`
* json/pickle: `dump`

So nvm, there isn't a really "pythonic" way to do it. I think MontePy just needs to be consistent

So I propose:

1. `write` means to write out the state in a way that MCNP understands
2. `dump` means to serialize to a file (JSON or pickle)

`write` usually takes a string as an argument, so I think `write_problem` is a good name.

Reply:
That was my exact train of thought.